src/client.rs

```diff
@@ -124,6 +143,7 @@ pub trait V7Methods {
     ) -> Result<reqwest::Response, reqwest::Error>;
     fn team(&self) -> &String;
     fn api_endpoint(&self) -> &str;
+    fn v2(&self) -> V7Client;
```

Comment:
**suggestion:** Instead of returning a new instance of a client with cloned internal values, could we add `api_version` back into the struct and have this method mutate the value?

e.g. `&mut self -> self.api_version = ApiVersion::V2`